### PR TITLE
Unified SLRO and LLRO into one LRO pattern.

### LINKED ISSUE #1737: Unify SLRO and LLRO into one LRO pattern
Previously we made a new design of [SLRO](https://gist.github.com/ShivangiReja/7949773c9759f9a597d21787afd02302) and implement for both [data plane and  mgmt. plane](https://github.com/Azure/autorest.csharp/pull/1473), and the [LLC](https://github.com/Azure/autorest.csharp/pull/1715) is also ongoing.

But to most of mgmt. sdk the LRO it really long, and data plan should be more complex because it's depend on the data size it process. 
After we revisit this feature, the architects decided that it is better to avoid picking a default between SLRO and LLRO for customers, so we will have one LRO pattern https://github.com/Azure/azure-sdk/issues/3662#issuecomment-985112894.

### PR DESCRIPTION
**Pull request by ArthurMa1978**
# Description

Previously we made a new design of [SLRO](https://gist.github.com/ShivangiReja/7949773c9759f9a597d21787afd02302) and implement for both [data plane and mgmt. plane](https://github.com/Azure/autorest.csharp/pull/1473), and the [LLC](https://github.com/Azure/autorest.csharp/pull/1715) is also ongoing.

But to most of mgmt. sdk the LRO it really long, and data plan should be more complex because it's depend on the data size it process.
After we revisit this feature, the architects decided that it is better to avoid picking a default between SLRO and LLRO for customers, so we will have one LRO pattern https://github.com/Azure/azure-sdk/issues/3662#issuecomment-985112894.

Closes #1737 

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [x] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.ResourceManager.Sample/Generated/AvailabilitySet.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ public virtual IEnumerable<Location> GetAvailableLocations(CancellationToken can
         /// <summary> Delete an availability set. </summary>
         /// <param name="waitForCompletion"> Waits for the completion of the long running operations. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public async virtual Task<AvailabilitySetDeleteOperation> DeleteAsync(bool waitForCompletion = true, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/AvailabilitySet.cs:155]: not final on this as the approach lets hold off on merging until we get confirmation

**Review by m-nash — CHANGES_REQUESTED** [review]

File: test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (AlexanderSher) [inline test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj:4]: What's the issue with net 6.0?

Comment (ArthurMa1978) [reply test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj:4]: Got error when open the sln in vs:
Error occurred while restoring NuGet packages: Invalid restore input. Invalid target framework 'unsupported'.

Comment (ArthurMa1978) [reply test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj:4]: It's happened in vs2019, vs2022 is ok, so do we need to compatible with vs2019?

Comment (AlexanderSher) [reply test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj:4]: Let's have one target for now.

**Review by m-nash — APPROVED** [review]

**Pull request merged**